fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -1688,12 +1732,77 @@ struct RocksDBKeyValueStore : IKeyValueStore {
+		readRangeLatencyHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READRANGE_LATENCY_HISTOGRAM, Histogram::Unit::microseconds);
+		readValueLatencyHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READVALUE_LATENCY_HISTOGRAM, Histogram::Unit::microseconds);
+		readPrefixLatencyHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READPREFIX_LATENCY_HISTOGRAM, Histogram::Unit::microseconds);
+		readRangeActionHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READRANGE_ACTION_HISTOGRAM, Histogram::Unit::microseconds);
+		readValueActionHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READVALUE_ACTION_HISTOGRAM, Histogram::Unit::microseconds);
+		readPrefixActionHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READPREFIX_ACTION_HISTOGRAM, Histogram::Unit::microseconds);
+		readRangeQueueWaitHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READRANGE_QUEUEWAIT_HISTOGRAM, Histogram::Unit::microseconds);
+		readValueQueueWaitHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READVALUE_QUEUEWAIT_HISTOGRAM, Histogram::Unit::microseconds);
+		readPrefixQueueWaitHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READPREFIX_QUEUEWAIT_HISTOGRAM, Histogram::Unit::microseconds);
+		readRangeNewIteratorHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READRANGE_NEWITERATOR_HISTOGRAM, Histogram::Unit::microseconds);
+		readValueGetHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READVALUE_GET_HISTOGRAM, Histogram::Unit::microseconds);
+		readPrefixGetHistogram = Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE, ROCKSDB_READPREFIX_GET_HISTOGRAM, Histogram::Unit::microseconds);
+
+		actors.push_back(updateHistogram(this, latencyMeasure));
 		writeThread->addThread(
-		    new Writer(db, defaultFdbCF, id, readIterPool, perfContextMetrics, SERVER_KNOBS->ROCKSDB_READ_PARALLELISM),
+		    new Writer(db, defaultFdbCF, id, readIterPool, perfContextMetrics, SERVER_KNOBS->ROCKSDB_READ_PARALLELISM, latencyMeasure),
 		    "fdb-rocksdb-wr");
 		TraceEvent("RocksDBReadThreads").detail("KnobRocksDBReadParallelism", SERVER_KNOBS->ROCKSDB_READ_PARALLELISM);
 		for (unsigned i = 0; i < SERVER_KNOBS->ROCKSDB_READ_PARALLELISM; ++i) {
-			readThreads->addThread(new Reader(db, defaultFdbCF, readIterPool, perfContextMetrics, i), "fdb-rocksdb-re");
+			readThreads->addThread(new Reader(db, defaultFdbCF, readIterPool, perfContextMetrics, i, latencyMeasure), "fdb-rocksdb-re");
+		}
+	}
+
+	ACTOR Future<Void> updateHistogram(RocksDBKeyValueStore* kvStore, ThreadReturnPromiseStream<std::tuple<int, std::string, double>>* latencyMeasure) {
```

Comment:
Please use FutureStream instead of PromiseStream. PromiseStream should not be accessed by main threads.